### ISSUE TITLE
Rename SampledSpanStore to LocalSpanStore.

### ISSUE THREAD

**Issue by bogdandrutu** [census-instrumentation/opencensus-java#700]
Add an option in the SpanBuilder to force the newly created span to save "samples". This is equivalent with users registered in advance that span name using the SampledSpanStore.registerSpanNamesForCollection method.

This helps gRPC to avoid a dependency on opencensus in the auto generated code.

/cc @zhangkun83

**Comment by adriancole** [comment]
Iiuc what it does I think SampledSpanStore isnt entirely intuitive name as
it means it will be available for query locally. This affects naming of
this option.

**Comment by adriancole** [comment]
So here is a high level proposal.

The idea that the local span store contains sampled data isnt imho important enough to get into the name. Rather, the fact that it is local/in memory is.

Ex LocalSpanStore resonates a bit more than sampled as the word samples also applies to exported data

LocalSampler.passSpansNamed(foo) or similar function could exist as well a spanbuilder.localSampler(ALWAYS) or such..

In other words feels more intuitive to me that if we have multiple sampling things we qualify by type ex ExportSampler or LocalSampler.. might make features easier to reason with or... only in my head!

Anyway hope this sparks something useful

**Comment by bogdandrutu** [comment]
I like the LocalSpanStore name, is probably better.

So to summarize what we will do:
1. Add SpanBuilder.setRegisterNameForLocalSpanStore
2. Rename SampledSpanStore tp LocalSpanStore.

Does that sound better?

**Comment by adriancole** [comment]
>
> I like the LocalSpanStore name, is probably better.
>
> So to summarize what we will do:
>
>    1. Add SpanBuilder.setRegisterNameForLocalSpanStore
>    2. Rename SampledSpanStore tp LocalSpanStore.
>
> Does that sound better?
>
yeap I like it better :) thank you!

**Comment by bogdandrutu** [comment]
Changed the name of the issue because the initial thing was solved there is only one remaining item.

**Comment by bogdandrutu** [comment]
This is probably obsolete now with https://github.com/census-instrumentation/opencensus-java/issues/981

**Comment by HailongWen** [comment]
Should we still rename it in #981, or just leave it as it be? @bogdandrutu

**Comment by bogdandrutu** [comment]
We should do that, but the class will become internal to the zpages package with that change.
